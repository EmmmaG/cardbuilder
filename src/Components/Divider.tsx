import * as React from 'react'
import { withStyles, WithStyles, Theme } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import DividerComponent from '@material-ui/core/Divider'

const styles = (theme: Theme) => ({
    title: {
        color: theme.palette.grey[400],
        fontWeight: 600,
    },
    container: {
        marginBottom: '25px',
    },
})

export interface Props extends WithStyles<typeof styles> {
    title: string
}

export class Divider extends React.Component<Props> {
	public render() {
		const { classes, title } = this.props

		return (
			<div className={classes.container}>
                <Typography className={classes.title}>{title}</Typography>
                <DividerComponent />
            </div>
		)
	}
}

export default withStyles(styles)(Divider)
