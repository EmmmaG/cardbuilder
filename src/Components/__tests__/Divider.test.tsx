import * as React from 'react'
import { shallow } from 'enzyme'
import idObj from 'identity-obj-proxy'

import Divider, { Props } from '../Divider'

const props: Props = {
    title: 'Details',
	classes: idObj,
}

describe('Divider', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<Divider {...props} />).dive()
		expect(wrapper).toMatchSnapshot()
	})
})
