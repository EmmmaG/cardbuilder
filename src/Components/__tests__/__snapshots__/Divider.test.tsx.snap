// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider renders correctly 1`] = `
<div
  className="Divider-container-2"
>
  <WithStyles(Typography)
    className="Divider-title-1"
  >
    Details
  </WithStyles(Typography)>
  <WithStyles(Divider) />
</div>
`;
